### PR TITLE
Some improvements for chaincode languages and metrics

### PR DESCRIPTION
**Pull request by jeffywu**
There are a few improvements in this PR:
* Documented a resolution to the EKS issue in the FAQ.
* Changed parts of the chaincode workflow to support golang
* Allowed different language chaincode to be deployed in one network.yaml file
* Added the ability to configure metrics for peers and orderers

I used release/0.7 as a base for the changes, I wasn't sure if I could use some of the changes on master because of the patches. I can rebase these changes to a different branch if you think that makes sense.

### CODE REVIEW

**Comment by raftAtGit** [comment]
Thanks! I’m at vacation so this need to wait a few weeks for inspecting and merging.

**Comment by gWOLF3** [comment]
Nice implementation to support statsD as well.  @jeffywu 

I am also preparing a PR to add support for built in heathchecks (readiness/liveness) added in 1.4.1. I added in support for prometheus but like how you added statsD compatibility. Working over an issue with the orderer healthchecks right now.

**Comment by jeffywu** [comment]
I didn't test it on master, I've only tested this on the release/0.7 tag.

File: fabric-kube/chaincode-flow/templates/chaincodes-workflow.yaml [hunk 1]
```diff
@@ -241,20 +241,28 @@ spec:
...
     # peer: {{ $peer }}
     inputs:
     retryStrategy:
       limit: {{ $.Values.retryCount }}
     container:
-      image: hyperledger/fabric-peer:{{ $.Values.hyperledgerVersion }}
       # chaincode is packed with tar, first extract it
+      {{- if eq $chaincode.language "golang" }}
+      image: hyperledger/fabric-tools:{{ $.Values.hyperledgerVersion }}
+      command: ["sh", "-c", "mkdir -p $GOPATH/src/chaincode && \
```
Comment (raftAtGit) [inline fabric-kube/chaincode-flow/templates/chaincodes-workflow.yaml]: why copying chaincode source to $GOPATH? is this required?

File: fabric-kube/chaincode-flow/values.yaml [hunk 2]
```diff
@@ -43,9 +43,4 @@ flow:
     # should we run invoke step? 
     # we invoke chaincode after instantiate/upgrade to force the peers to actually create the chaincode containers
     enabled: true 
-    parallel: true
-    function: '{"function":"ping","Args":[""]}'
-
-chaincode:
-  version: "1.0"
```
Comment (raftAtGit) [inline fabric-kube/chaincode-flow/values.yaml]: why removed global chaincode version and language? I think it's handy to have it global and make overridable by individual chaincodes

File: fabric-kube/chaincode-flow/templates/chaincodes-workflow.yaml (hunk 1 above)
Comment (jeffywu) [reply fabric-kube/chaincode-flow/templates/chaincodes-workflow.yaml]: Yes, this is required for golang chaincode: https://stackoverflow.com/questions/48030138/chaincode-installation-on-peer-in-hyperledger-fabric

File: fabric-kube/chaincode-flow/values.yaml (hunk 2 above)
Comment (jeffywu) [reply fabric-kube/chaincode-flow/values.yaml]: That makes sense, I can look at adding that back in.

**Comment by jeffywu** [comment]
Added global language and version overrides.

**Comment by sillysachin** [comment]
Is this integrated?

**Pull request closed without merging**